File: model/src/main/java/io/sparkled/model/entity/SongAudio.java
package io.sparkled.model.entity;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "song_audio")
public class SongAudio {

    @Id
    @Column(name = "song_id")
    private Integer songId;

    @Lob
    @Column(name = "audio_data")
    private byte[] audioData;

    public Integer getSongId() {
        return songId;
    }

    public SongAudio setSongId(Integer songId) {
        this.songId = songId;
        return this;
    }

    public byte[] getAudioData() {
        return audioData;
    }

    public SongAudio setAudioData(byte[] audioData) {
        this.audioData = audioData;
        return this;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        SongAudio audio = (SongAudio) o;
        return Objects.equals(songId, audio.songId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(songId);
    }

    @Override
    public String toString() {
        return "SongAudio{" +
                "songId=" + songId +
                '}';
    }
}

File: model/src/main/kotlin/io/sparkled/model/entity/SongAudio.kt
package io.sparkled.model.entity

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.Lob
import javax.persistence.Table

@Entity
@Table(name = "song_audio")
class SongAudio {

    @Id
    @Column(name = "song_id")
    private var songId: Int? = null

    @Lob
    @Column(name = "audio_data")
    private var audioData: ByteArray? = null

    fun getSongId(): Int? {
        return songId
    }

    fun setSongId(songId: Int?): SongAudio {
        this.songId = songId
        return this
    }

    fun getAudioData(): ByteArray? {
        return audioData
    }

    fun setAudioData(audioData: ByteArray): SongAudio {
        this.audioData = audioData
        return this
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as SongAudio

        if (songId != other.songId) return false

        return true
    }

    override fun hashCode(): Int {
        return songId ?: 0
    }

    override fun toString(): String {
        return "SongAudio(songId=$songId)"
    }
}
